### PR TITLE
[6517] Add a single item per line view to search and browse item selection dialogs

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6517

### CODE REVIEW

**Comment by jvega190** [comment]
Some decisions pending

File: ui/app/src/utils/state.ts [hunk 1]
```diff
@@ -359,6 +360,20 @@ export function getStoredBrowseDialogCompactMode(username: string): boolean {
...
+
+export function getStoredBrowseDialogViewMode(username: string): MediaCardViewModes {
+  // The viewMode field used to be compactMode (boolean). For backwards compatibility, if the viewMode is not set and
+  // compactMode exists, set value accordingly.
+  const backwardsCompatibilityValue = getStoredBrowseDialogCompactMode(username) ? 'compact' : 'card';
+  return (
+    (localStorage.getItem(`craftercms.${username}.browseDialog.viewMode`) as MediaCardViewModes) ??
+    backwardsCompatibilityValue
+  );
+}
```
Comment (rart) [inline ui/app/src/utils/state.ts]: Let's clean up.
- Remove getStoredBrowseDialogCompactMode
- Check for the old key in the new storing modes and if present delete and write the new.
  - The setted should delete the old and write the new
  - The getter must check the old and if not present go to the new
  - Add a clean up todo for later versions to fully remove old key
- Adjust the Account preference clean up stuff to include all this stuff too

File: ui/app/src/utils/state.ts [hunk 2]
```diff
@@ -359,6 +360,20 @@ export function getStoredBrowseDialogCompactMode(username: string): boolean {
...
 }
 
+export function setStoredBrowseDialogViewMode(username: string, mode: MediaCardViewModes): void {
+  localStorage.setItem(`craftercms.${username}.browseDialog.viewMode`, mode);
+}
+
+export function getStoredBrowseDialogViewMode(username: string): MediaCardViewModes {
+  // The viewMode field used to be compactMode (boolean). For backwards compatibility, if the viewMode is not set and
+  // compactMode exists, set value accordingly.
+  const backwardsCompatibilityValue = getStoredBrowseDialogCompactMode(username) ? 'compact' : 'card';
```
Comment (rart) [inline ui/app/src/utils/state.ts]:
```suggestion
  const backwardsCompatibilityValue: MediaCardViewModes = getStoredBrowseDialogCompactMode(username) ? 'compact' : 'card';
```

**Review by rart — CHANGES_REQUESTED** [review]
Make sure to test with data of various lengths (very short/long, normal), etc

**Review by rart — CHANGES_REQUESTED** [review]
From previous review:  Account preference clean up stuff to include all this stuff too

**Comment by jvega190** [comment]
> From previous review: Account preference clean up stuff to include all this stuff too

yeap, updated removeStoredBrowseDialogCompactMode to removeStoredBrowseDialogViewMode, which is called in the preferences clean up under AccountManagement/utils.tsx

**Review by rart — APPROVED** [review]

**Pull request merged**